lib/x11utils.pm

```diff
@@ -68,7 +68,17 @@ all possible options should be handled within loop to get unlocked desktop
 =cut
 sub ensure_unlocked_desktop {
     my $counter = 10;
-    while ($counter--) {
+
+    # check if the screenbuffer was not updated and record a softfailure in that case
+    if check_screen 'screenlock' {
+        send_key 'shift';
+        if check_screen 'screenlock' {
+            record_soft_failure 'bsc#1168979 screenbuffer containing old data';
+        }
+      }
```

Comment:
I doubt this can work reliably as without any waiting time for the effect of the key press this is likely to cause false alarms. Also I doubt we have any new regression for "screenbuffer containing old data" here unless you can back this assumption with very good statistics proofing that any related update in the build really causes a problem now that was not there in before.

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10064/files#diff-f7cb81ed3ad3075100bf95d110f7e8adL103 has more explanation for the rather old issue of "still old screen content" shown. Introducing a check at a second location in code is unlikely to work out correctly. If you insist on keeping this code then I suggest to safe-guard with an additional version check to not trigger likely false-alarms at least on product versions that can not be affected.

Reply:
@DrMullings, please always use `check_screen 'something', 0;` or `check_screen('something`, 0);` The timeout 0 is important.

Reply:
Actually the 0 is *not* important. It's the default since some months

Reply:
> https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10064/files#diff-f7cb81ed3ad3075100bf95d110f7e8adL103 has more explanation for the rather old issue of "still old screen content" shown. Introducing a check at a second location in code is unlikely to work out correctly. If you insist on keeping this code then I suggest to safe-guard with an additional version check to not trigger likely false-alarms at least on product versions that can not be affected.

Actually this reminds me somehow of a bug on aarch64 regarding virto-gpu not being able to properly switch back and forth on aarch64. So I think it's more of an architecture + version than just the version alone.

Reply:
As I state in the bug, it is reproducible on x86_64